smarts/core/road_map.py

```diff
@@ -96,13 +106,34 @@ def generate_routes(
         via: Optional[Sequence[RoadMap.Road]] = None,
         max_to_gen: int = 1,
     ) -> List[RoadMap.Route]:
-        """Routes will be returned in order of increasing length"""
+        """Generates routes between two roads. Routes will be returned in order of increasing
+         length.
+        Args:
+            start_road:
+                The beginning road of the generated routes.
+            end_road:
+                The end road of the generated routes.
+            via:
+                All edges that the generated routes must pass through.
+            max_to_gen:
+                The maximum number of routes to generate.
+        Returns:
+            A list of generated routes that satisfy the given restrictions.
```

Comment:
Suggestion:  I'd move "Routes will be returned in order of increasing length." here, down from the general description above.